vendor/engines/split_accounts/app/models/split_accounts/split_account.rb

```diff
@@ -36,10 +39,24 @@ def extra_penny_count
     #   subaccounts.where("expires_at IS NOT NULL").order(expires_at: :asc).first
     #
     def earliest_expiring_subaccount
-      subaccounts = splits.map{ |split| split.subaccount }.select(&:expires_at)
+      subaccounts = splits.map(&:subaccount).select(&:expires_at)
       subaccounts.sort_by(&:expires_at).first
     end
 
+    def outmost_suspended_subaccount
+      subaccounts = splits.map(&:subaccount).select(&:suspended_at)
+      subaccounts.sort_by(&:suspended_at).last
+    end
+
+    def set_expires_at_from_subaccounts
+      self.expires_at = earliest_expiring_subaccount.expires_at
```

Comment:
Technically expires_at should always be set.  But I like the idea of adding a guard clause to make this more robust for potential future enhancements or mocked tests.